File: LibDialog/src/main/java/com/ooftf/master/widget/dialog/ui/ListBlurDialog.java
package com.ooftf.master.widget.dialog.ui;

import android.app.Activity;

import com.ooftf.master.widget.dialog.delegate.ListDialogDelegate;
import com.ooftf.master.widget.dialog.inteface.DialogOnItemClickListener;
import com.ooftf.master.widget.dialog.inteface.ListDialogInterface;

import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * @author ooftf
 * @email 994749769@qq.com
 * @date 2018/10/17 0017
 */
public class ListBlurDialog extends BlurDialog implements ListDialogInterface {
    ListDialogDelegate delegate;

    public ListBlurDialog(@NotNull Activity activity) {
        super(activity);
        delegate = new ListDialogDelegate(this, activity);
    }


    @Override
    public ListDialogInterface setShowCancel(boolean showCancel) {
        return delegate.setShowCancel(showCancel);
    }

    @Override
    public ListDialogInterface setList(List<String> data) {
        return delegate.setList(data);
    }

    @Override
    public ListDialogInterface setOnItemClickListener(DialogOnItemClickListener listener) {
        return delegate.setOnItemClickListener(listener);
    }
}

File: lib-dialog/src/main/java/com/ooftf/master/widget/dialog/ui/ListBlurDialog.kt
package com.ooftf.master.widget.dialog.ui

import android.app.Activity
import android.widget.TextView
import com.ooftf.master.widget.dialog.delegate.ListDialogDelegate
import com.ooftf.master.widget.dialog.inteface.DialogOnItemClickListener
import com.ooftf.master.widget.dialog.inteface.ListDialogInterface

/**
 * @author ooftf
 * @email 994749769@qq.com
 * @date 2018/10/17 0017
 */
open class ListBlurDialog(activity: Activity) : BlurDialog(activity), ListDialogInterface {
    var delegate: ListDialogDelegate = ListDialogDelegate(this, activity)
    override fun setItemViewAdapter(adapter: ((textView: TextView, position: Int) -> Unit)): ListBlurDialog {
        delegate.setItemViewAdapter(adapter)
        return this
    }

    override fun setShowCancel(showCancel: Boolean): ListBlurDialog {
        delegate.setShowCancel(showCancel)
        return this
    }

    override fun setList(data: List<String>): ListBlurDialog {
        delegate.setList(data)
        return this
    }

    override fun setOnItemClickListener(listener: DialogOnItemClickListener): ListBlurDialog {
        delegate.setOnItemClickListener(listener)
        return this
    }

}